### PR TITLE
Check presence of multi-types before validating new mapping

### PR DESCRIPTION
**Pull request by ywelsch**
Before doing any kind of validation on a new mapping, we should first do the multi-type validation in order to provide better error messages. For #29313, this means that the exception message will be
`Rejecting mapping update to [range_index_new] as the final mapping would have more than 1 type: [_doc, mytype]`
instead of 
`[expected_attendees] is defined as an object in mapping [mytype] but this name is already used for a field in other types"`

### CODE REVIEW

**Review by jpountz — APPROVED** [review]
Thanks @ywelsch.

File: server/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexIT.java [hunk 1]
```diff
@@ -264,25 +264,6 @@ public void onFailure(Exception e) {
         logger.info("total: {}", expected.getHits().getTotalHits());
     }
 
-    /**
-     * Asserts that the root cause of mapping conflicts is readable.
-     */
-    public void testMappingConflictRootCause() throws Exception {
```
Comment (ywelsch) [inline server/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexIT.java]: while backporting to 6.x, I've reenabled this test (but set `index.version.created` to 5.x)

**Comment by dpursehouse** [comment]
Queries that were working against 6.3.2 are now failing against 6.4.0 with the message `...as the final mapping would have more than 1 type...`, but neither this PR (#29316) nor the one it references (#29313) are mentioned as breaking changes in the [release notes](https://www.elastic.co/guide/en/elasticsearch/reference/current/release-notes-6.4.0.html).

**Comment by dpursehouse** [comment]
Sorry, disregard my previous comment.  It was my mistake.

**Comment by ebuildy** [comment]
Why this is loggued with ``DEBUG`` level ?

```
[2020-04-16T12:40:57,790][DEBUG][o.e.a.b.TransportShardBulkAction] [events-2020.04.16][0] failed to execute bulk item (index) BulkShardRequest [[events-2020.04.16][0]] containing [index {[events-2020.04.16][doc][uWoBg3EBC9-1w63Yrl-y], source....]
java.lang.IllegalArgumentException: Rejecting mapping update to [events-2020.04.16] as the final mapping would have more than 1 type: [_doc, doc]
	at org.elasticsearch.index.mapper.MapperService.internalMerge(MapperService.java:408) ~[elasticsearch-6.3.2.jar:6.3.2]
	at org.elasticsearch.index.mapper.MapperService.internalMerge(MapperService.java:356) ~[elasticsearch-6.3.2.jar:6.3.2]
	at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:288) ~[elasticsearch-6.3.2.jar:6.3.2]
```

This should be an ERROR isnit?

**Pull request merged**